### ISSUE TITLE
LibConError

### ISSUE THREAD

**Issue by lucidnx** [joedf/Qonsole#11]
Hello, when i run qonsole, open it with hotkey, I will get error:

window title: LibConError
AttachConsole() Failure
Errorlevel: 0
A_LastError: 6
Will now Exit.

i have options: abort, retry, ignore.
When try to click on retry, console appears normally as expected, then it is working fine until closing and re-opening qonsole

windows 10, build: 17763.618

**Comment by joedf** [comment]
Hey there! Thanks for reporting this. That tends to happen when the console is delayed or failed to start.
Are you consistently getting this error?

Also, what are your settings?

**Comment by lucidnx** [comment]
yes, it's on every start of console and triggering it with hotkey. setup:
button: F12
transparency 5%
w 1440
h 400
centered, bottom
no startup arguments
vertical offset 0
animation speed 1, delay 0, dx 25
bg color 1A1A1A, transparency 100 speed 100
run qonsole on start,
ctrl+v pasting
reduce memory usage
i am starting powershell instead of cmd

I have also found that on earlier version it was not happening

**Comment by joedf** [comment]
Interesting, could you send me the last version number where this issue does not occur?
Thanks 👍

**Comment by lucidnx** [comment]
on 1.4.2 it was OK i think, now in 1.4.5 it's throwing error :)

**Comment by joedf** [comment]
Great, thanks. I'll look into it. 🤓

**Comment by joedf** [comment]
Hmm... I just looked at the source of v1.4.5 and v1.4.2. There is no difference other than the High DPI scaling support I've added. I could find anything related to attaching to console... 
This might be related to issue #8 ... 
Do you by chance have the settings file you used for both v1.4.2 and 1.4.5 ? 😀

**Comment by lucidnx** [comment]
well, I only replaced exe file. but it's happening now with this settings, what does it mean? that error? I think that it have something to do with console open time. (powershell load take longer than regular cmd) and qonsole assume it's running before it really is? also when i type exit, next time i press hotkey, it show error too.

**Comment by joedf** [comment]
I think you are right with the timing issue. I'll make a custom build later tonight. And you can let me know if it works better or not. 👍

**Comment by joedf** [comment]
try this? [Qonsole_v1.4.5c.zip](https://github.com/joedf/Qonsole/files/3479671/Qonsole_v1.4.5c.zip)
I did simple increase for time waited for the cmd window to appear to 3 seconds.

**Comment by lucidnx** [comment]
thanks, but it's not fixed.. also when I press Ignore error, it will cycle some command errors:
AttachConsole Failure,
getFontSize Failure,
getConsoleSize Failure,
getConsoleHandle Failure,
setConsoleSize Failure

**Comment by joedf** [comment]
If you try cmd instead of powershell?

**Comment by lucidnx** [comment]
no, it's same for cmd and powershell, but if i quickly hit hotkey after reloading qonsole, error is not shown. but only sometimes. I think that I need to hit hotkey in 0.5s accurately :D

**Comment by joedf** [comment]
Hmmm I cant reproduce this reliably. Could you try with Console2?
https://sourceforge.net/projects/console/files/console-devel/2.00/Console-2.00b148-Beta_32bit.zip/download

**Comment by lucidnx** [comment]
Sure, it it help: I am on Windows 10 Pro 64bit, I have installed Hyper-V, some additional windows packages like SMBv1 support, and other networking and IT oriented things, WSL (Linux subsystem), OpenVPN... Maybe Hyper-V and WSL can affect it?

**Comment by joedf** [comment]
It could be WSL. Either way, there is a lag causing the attach function to fail.
I'm considering a rewrite (possibly in C#) and cutting WinXP support, have Win7 as a minimum.

**Comment by lucidnx** [comment]
Well, windows XP is deprecated and I don't know person who is using it as IT guy :)

**Comment by joedf** [comment]
Good point 😅

**Comment by lucidnx** [comment]
Maybe you can implement switch for cmd/powershell/bash(wsl) with specific linux you choose in reworked version ;)

**Comment by joedf** [comment]
They all use the same new console as far as I know.
https://docs.microsoft.com/en-us/previous-versions/orphan-topics/ws.11/mt427362(v=ws.11)
I haven't experimented much with WSL much but it should all be pretty much the same in terms of handling the console window. I will take note of this suggestion for the rewritten version. thanks 👍

**Comment by BlueDrink9** [comment]
If it's a lagging issue, and this is ahk, can I (naïvely) suggest something like a `winwait`or `waitwinactive` call?

**Comment by joedf** [comment]
Thanks 👍 
I already have a `WinWaitActive` but perhaps it never gets 'Active'. So maybe I'll try `winwait` instead. If not, maybe an asynchronous rewrite of the initialization part would mitigate this.
